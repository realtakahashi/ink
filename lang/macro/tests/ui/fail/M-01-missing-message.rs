use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod missing_message {
    #[ink(storage)]
    struct MissingMessage {}

    impl MissingMessage {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }
    }
}

fn main() {}
