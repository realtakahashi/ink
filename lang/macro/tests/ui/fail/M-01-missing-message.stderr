error: ink! contracts require at least one `#[ink(message)]`
  --> $DIR/M-01-missing-message.rs:4:1
   |
4  | / mod missing_message {
5  | |     #[ink(storage)]
6  | |     struct MissingMessage {}
7  | |
...  |
13 | |     }
14 | | }
   | |_^
